dev/set_matrix.py

```diff
@@ -195,7 +195,11 @@ def get_changed_flavors(changed_files, flavors):
     ['pytorch', 'xgboost']
     >>> get_changed_flavors(["mlflow/xgboost.py"], flavors)
     ['xgboost']
-    >>> get_changed_flavors(["tests/xgboost/test_xgboost_autolog.py"], flavors)
+    >>> get_changed_flavors(["tests/xgboost/test_xxx.py"], flavors)
+    ['xgboost']
+    >>> get_changed_flavors(["tests/xgboost_autolog/test_xxx.py"], flavors)
+    ['xgboost']
+    >>> get_changed_flavors(["tests/xgboost_autologging/test_xxx.py"], flavors)
     ['xgboost']
     >>> get_changed_flavors(["README.rst"], flavors)
```

Comment:
Added new test cases to ensure the new regular expression captures a flavor name properly.

Reply:
Locally ran:

```sh
python dev/set_matrix.py --changed-files tests/xgboost_autolog/test_xxx.py
```

and got this (correct) result:

```
{
  "job_name": [
    "xgboost / 0.90 / autologging",
    "xgboost / 0.90 / models",
    "xgboost / 1.0.2 / autologging",
    "xgboost / 1.0.2 / models",
    "xgboost / 1.1.1 / autologging",
    "xgboost / 1.1.1 / models",
    "xgboost / 1.2.1 / autologging",
    "xgboost / 1.2.1 / models",
    "xgboost / 1.3.0.post0 / autologging",
    "xgboost / 1.3.0.post0 / models",
    "xgboost / dev / autologging",
    "xgboost / dev / models"
  ],
  ...
```